cmd/main.go

```diff
@@ -27,13 +28,26 @@ func init() {
 	flag.BoolVar(&config.NoTeamsAPI, "noteamsapi", false, "Disable all access to the teams API")
 	flag.Parse()
 
+	cmd := exec.Command("cat", "/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	operatorNamespaceBytes, err := cmd.Output()
+	if err != nil {
+		log.Fatalf("Unable to detect operator namespace from within its pod  due to %v", err)
+	}
+
 	configMapRawName := os.Getenv("CONFIG_MAP_NAME")
 	if configMapRawName != "" {
-		err := config.ConfigMapName.Decode(configMapRawName)
+
+		operatorNamespace := string(operatorNamespaceBytes)
+		namespacedConfigMapName := operatorNamespace + "/" + configMapRawName
```

Comment:
Seems to be some unnecessary work. I'd rather introduce a DecodeWithDefaultNamespace function in spec/types.go, that receives 2 arguments, a value, and a namespace to fill-in as a default one. The logic from Decode will be moved there, and the Decode itself will call DecodeWithNamespace(value, v1.NamespaceDefault). The code in main will call DecodeWithDefaultNamespace directly, supplying the operator namespace as a second option.

Reply:
this is the only place where `Decode` is actually used; right now i am unsure if it is sensible to split it into 2 functions.

Reply:
Nope, AFAIK Decode implements an interface and it used, for instance, when Decoding values from JSON in the operator configuration (i.e. OAuthTokenSecretName or InfrastructureRolesSecretName).

Reply:
Actually, given that we likely want those objects to default to the operator namespace as well, maybe it makes sense to just change the Decode to use the operator namespace instead of v1.DefaultNamespace.

Reply:
good point, will address in the commit that makes operator fetch manifests from the namespace it is deployed into